### DISCUSSION TITLE
Cache HTML codes inside Twig

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#47426] [Q&A] [answered]
Hi
Do we need to cache HTML codes inside Twig? Does it have any effect?
Some people say that Full Page Cache is better than cache some sections of page, is it correct?
Here we can cache html inside twig:
https://twig.symfony.com/doc/3.x/tags/cache.html

**Comment by GromNaN** [comment, marked as answer]
Caching is useful when the cost of reading the cache is lower than the cost of generating the cached contents. Cost being CPU, memory and network usage, latency and availability.

Is the context of Twig, we tend to fetch and process the data before calling the render. Twig is a function that convert an array of data into a HTML string. Using a cache for this would be counterproductive.

Sometime, some templates are costly to generate: they are inherently complex or they use helpers that make network requests. It might be useful to use the `cache` tag to soften this cost.

Caching data queries or HTTP responses is usually a lot more efficient.
